crates/dojo-lang/src/db.rs

```diff
@@ -65,3 +69,24 @@ impl DojoRootDatabaseBuilderEx for RootDatabaseBuilder {
         self.with_dojo(config.content.world)
     }
 }
+
+pub fn update_crate_roots_from_metadata(
```

Comment:
Note: This code is going to be iterated multiple times I think, there are rought edges, like what should happen if there are multiple targets in a package, or what if there's no `scarb` binary in `PATH`. Keep an eye on it :)